src/gfx/gfx_wgpu.cpp

```diff
@@ -0,0 +1,116 @@
+#include "gfx_wgpu.hpp"
+#include "stdio.h"
+
+#ifdef __EMSCRIPTEN__
+#include <emscripten/html5.h>
+#endif
+static void tickAsyncRequests() {
+#ifdef __EMSCRIPTEN__
+  printf("tickAsyncRequests\n");
+  emscripten_sleep(1);

```

Comment:
The only place I need this currently is when requesting the webgpu adapter on browsers, since it has a callback.

I could move it up if you prefer, but in that case the GFX.Window would delay until the adapter is there

Reply:
It's more about `emscripten_sleep`, it's a wrapper to some async features that we already use directly via scheduler/coroutines. Can't you just use CB_SUSPEND? It would do the same in a better way.

Reply:
I removed this from the context code, requesting device/adapter runs async now so on web it just blank screens until it either gets one or fails during activate.